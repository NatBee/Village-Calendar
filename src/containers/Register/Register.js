import React, { Component } from 'react';
import './Register.css';
import { createNewCalendar, addUsersToCalendar } from '../../helper/apiCall';
import { setCalendarID } from '../../actions/index';
import { connect } from 'react-redux';

class Register extends Component {

  createCalendar = async () => {
    const newCalendarID = await createNewCalendar();
    await this.props.setCalendarID(newCalendarID);
  }

  addUsers = async () => {
    // console.log('hi')
    await addUsersToCalendar(this.props.calendarID )
  }

  render() {
    return(
      <div>
        <h1>Welcome to Village</h1>
        <p>Follow these step to setup your calendar:</p>
          <ol>
            <li>Click the Google Log In Button above to login or create the google account you want attached to your Village Calendar</li>
            <li>Create Village Calendar</li>
            <button onClick={this.createCalendar}>Create Calendar</button>
            <li>Add people to your village</li>
            <input type='text' placeholder='friend@gmail.com'/>
            <button onClick={this.addUsers}>Add Users</button>
          </ol>
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  calendarID: store.calendarID
})

export const mapDispatchToProps = (dispatch) => ({
  setCalendarID: (calendarID) => dispatch(setCalendarID(calendarID))
})

export default connect(mapStateToProps, mapDispatchToProps)(Register);