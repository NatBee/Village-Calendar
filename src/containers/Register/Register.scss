.register {
  width: 300px;
  display: inline-block;
  background-color: #A7B9AB;
  opacity: .85;
  padding: 50px 40px 36px 40px;
  margin: 160px;
}

.register h2 {
  font-size: 1.25em;
  padding: 0 30px;
  margin: 0 0 45px;
  color: #3E4D4E;
}

.register h3 {
  margin: 25px 0 20px;
  color: #3E4D4E
}

.register button { 
  background-color: #3E4D4E;
  height: 30px;
  color: #DBE3D4;
  font-family: 'Exo', sans-serif;
  border: 2px solid #293435;
  margin: 0 0 5px;
  width: 70px;
}

.register button:hover {
  background-color: #293435;
}

.new-calendar h4 {
  margin: 0;
  color: #DBE3D4;
}

.new-calendar form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.new-calendar form input {
  margin: 10px 0 0px;
  font-size: 1em;
  padding: 2px 6px;
  width: 275px;
}

.new-calendar form button {
  width: 100px;
}

.new-calendar form .add-btn {
  margin: 25px 0px 0 45px;
}

.new-calendar form .go-calendar {
  margin: 25px 45px 0 0px;
}

.btn {
  display: flex;
  justify-content: space-around;
}