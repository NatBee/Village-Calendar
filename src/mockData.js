export const mockData = {
  user: {
    user: 'Dk', 
    credential: 'Xg', 
    additionalUserInfo: 'Pg', 
    operationType: "signIn"
  },
  events: [
    {
      start: "2018-02-09T09:00:00-07:00",
      end: "2018-02-09T10:00:00-07:00",
      title: "Super Fun Party",
      description: "Fun with friends",
      eventID: "7js6e24fo0kin4qcotoegu37hg",
      location: 'Mi Casa'
    }
  ],
  token: 'token_number',
  calendarID: 'calendarID',
  village: [{natalie: 'nataliesmontoya@gmail.com'}],
  time: {
    startTime: "2018-02-28T19:00:00.000Z", 
    endTime: "2018-02-28T20:00:00.000Z"
  },
  event: {
    start: "2018-02-09T09:00:00-07:00",
    end: "2018-02-09T10:00:00-07:00",
    title: "Super Fun Party",
    description: "Fun with friends",
    eventID: "7js6e24fo0kin4qcotoegu37hg",
    location: 'Mi Casa'
  }
}