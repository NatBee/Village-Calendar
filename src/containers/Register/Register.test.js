/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { Register, mapStateToProps, mapDispatchToProps } from './Register';
import { mockData } from '../../mockData';

describe('Register', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Register/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should map to the store correctly', () => {
    const mockStore = {
      calendarID: mockData.calendarID,
      village: mockData.village,
      token: mockData.token  
    }
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  })

  it('should call the dispatch function when using setCalendarID function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setCalendarID();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using addPeopleToVillage function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addPeopleToVillage();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should have a default state', () => {
    const wrapper = shallow(<Register/>);
    const expected = {
      name: '',
      email: '',
      newCalendar: false
    }
    expect(wrapper.state()).toEqual(expected);
  })
})