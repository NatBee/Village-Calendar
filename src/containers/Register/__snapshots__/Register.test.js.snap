// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h2>
          Would you like to load an existing calendar or create a new calendar?
        </h2>
        <div
          className="btn"
        >
          <button
            onClick={[Function]}
          >
            Existing
          </button>
          <button
            onClick={[Function]}
          >
            New
          </button>
        </div>
      </div>,
      "className": "register",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Would you like to load an existing calendar or create a new calendar?
          </h2>,
          <div
            className="btn"
          >
            <button
              onClick={[Function]}
            >
              Existing
            </button>
            <button
              onClick={[Function]}
            >
              New
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Would you like to load an existing calendar or create a new calendar?",
          },
          "ref": null,
          "rendered": "Would you like to load an existing calendar or create a new calendar?",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                Existing
              </button>,
              <button
                onClick={[Function]}
              >
                New
              </button>,
            ],
            "className": "btn",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Existing",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Existing",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "New",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h2>
            Would you like to load an existing calendar or create a new calendar?
          </h2>
          <div
            className="btn"
          >
            <button
              onClick={[Function]}
            >
              Existing
            </button>
            <button
              onClick={[Function]}
            >
              New
            </button>
          </div>
        </div>,
        "className": "register",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Would you like to load an existing calendar or create a new calendar?
            </h2>,
            <div
              className="btn"
            >
              <button
                onClick={[Function]}
              >
                Existing
              </button>
              <button
                onClick={[Function]}
              >
                New
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Would you like to load an existing calendar or create a new calendar?",
            },
            "ref": null,
            "rendered": "Would you like to load an existing calendar or create a new calendar?",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  onClick={[Function]}
                >
                  Existing
                </button>,
                <button
                  onClick={[Function]}
                >
                  New
                </button>,
              ],
              "className": "btn",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Existing",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Existing",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "New",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
